modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/validation/JavaValueProviderModelValidator.java

```diff
@@ -161,9 +168,24 @@ private void validateOptionsResolver(ParameterModel param, boolean mustBeStringT
       } else {
         if (parameterInfo.getExtractionExpression().equals(parameterInfo.getParameterName())) {
           MetadataType metadataType = allParameters.get(parameterInfo.getParameterName());
-          Class<?> expectedType = getType(metadataType)
-              .orElseThrow(() -> new IllegalStateException(format("Unable to get Class for parameter: %s",
-                                                                  parameterInfo.getParameterName())));
+          Set<TypeAnnotation> annotations = metadataType.getAnnotations();
+          Class<?> expectedType;
+          if (annotations.stream().anyMatch(ann -> ann instanceof EnumAnnotation)
```

Comment:
very uneficient assertion. If you want to test for both both annotations being present, simply fetch them with MetadataType.getAnnotation(Class)